tsconfig.base.json

```diff
@@ -0,0 +1,35 @@
+{
+  "compilerOptions": {
+    "allowSyntheticDefaultImports": true,
+    "jsx": "react-jsx",
+    "jsxImportSource": "preact",
+    "module": "es2015",
+    "moduleResolution": "node",
+    "noFallthroughCasesInSwitch": true,
+    "noImplicitAny": false,
+    "noImplicitReturns": true,
+    "noUnusedLocals": false,
+    "noUnusedParameters": false,
+    "resolveJsonModule": true,
+    "skipLibCheck": true,
+    "strict": true,
+    "target": "ESNEXT",
+    "composite": true,
+    "incremental": true,
+    "noEmit": false,
+    "declaration": true,
+    "emitDeclarationOnly": true,
+    "paths": {
+      "dinero.js": ["./packages/dinero.js/src/index"],
+      "@dinero.js/core": ["./packages/core/src/index"],
+      "@dinero.js/currencies": ["./packages/currencies/src/index"],
+      "@dinero.js/calculator-bigint": [
+        "./packages/calculator-bigint/src/index"
+      ],
+      "@dinero.js/calculator-number": [
+        "./packages/calculator-number/src/index"
+      ],
+      "test-utils": ["./test/utils/index"]
```

Comment:
One thing we do in [`algolia/instantsearch.js`](https://github.com/algolia/instantsearch.js) is use prefixes for internal dependencies (private packages and references) to avoid any issue with existing packages of the same name and make it more obvious when reading the code that this is internal.

What do you think we rename `test-utils` to `@dinero.js/test`?

Reply:
I can definitely do that, but that was the original original alias I made 😄

https://github.com/dinerojs/dinero.js/pull/666#discussion_r1015837973

Reply:
Do you want to just make it an external package?

Projects commonly do that with their base tsconf.json too. So it's easier to reference rather than long relative paths.

Reply:
HAHAHA that's a Spiderman meme situation for sure 😆

Let's keep it as it is and change that in a separate PR if we still want to.